pyfar/dsp/regularization.py

```diff
@@ -0,0 +1,313 @@
+    function is also passed.
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>> import matplotlib.pyplot as plt
+        >>>
+        >>> # get regularization object
+        >>> Regu = pf.dsp.RegularizedSpectrumInversion.from_frequency_range([20, 15e3])
+        >>> # create linear sweep and invert it:
+        >>> sweep = pf.signals.linear_sweep_time(1000, (20, 20e3))
+        >>> inv = Regu.invert(sweep)
+        >>>
+        >>> # design bandpass target function
+        >>> bp = pf.dsp.filter.butterworth(None, 4, (20, 15e3), 'bandpass', 44.1e3)
+        >>> target = bp.process(pf.signals.impulse(1000))
+        >>> # get regularization object with target function
+        >>> Regu_target = pf.dsp.RegularizedSpectrumInversion.from_frequency_range([20, 15e3])
+        >>> # invert signal
+        >>> inv_target = Regu_target.invert(sweep, target=target)
+        >>>
+        >>> # plot results
+        >>> pf.plot.freq(sweep, label="Original signal")
+        >>> pf.plot.freq(inv, label="Regularized inversion")
+        >>> pf.plot.freq(inv_target, label="Regularized inversion with target")
+        >>> plt.legend()
+
+    References
+    ----------
+    .. [1]  S. G. Norcross, M. Bouchard, and G. A. Soulodre, “Inverse Filtering
+            Design Using a Minimal-Phase Target Function from Regularization,”
+            Convention Paper 6929 Presented at the 121st Convention, 2006
+            October 5–8, San Francisco, CA, USA.
+    """ # noqa: E501
+
+    def __init__(self, regu_type: str="") -> None:
+
+        self._regu_type = regu_type
+
+        # throw error if object is instanced without classmethod
+        if regu_type == "":
+            raise RuntimeError("Regularization objects must be created "
+                             "using a classmethod.")
+
+    def __repr__(self):
+        return f"Regularization object with regularization "\
+            f"of type '{self._regu_type}'."
+
+    @classmethod
+    def from_frequency_range(cls, frequency_range):
+        r"""
+        Regularization from frequency range.
+        Defines a frequency range within which the regularization factor is set
+        to ``0``. Outside the frequency range the regularization factor is
+        ``1`` and can be scaled using the ``beta`` parameter.
+        The regularization factors are cross-faded using a raised cosine
+        window function with a width of :math:`\sqrt{2}f` above and below the
+        given frequency range.
+
+        Parameters
+        ----------
+        frequency_range : array, tuple
+            Tuple with two values for the lower and upper frequency limit.
+        """
+        instance = cls(regu_type = "frequency range")
+
+        if len(frequency_range) != 2:
+            raise ValueError(
+                "The frequency range needs to specify lower and upper limits.")
+
+        instance._frequency_range = np.asarray(frequency_range)
+
+        return instance
```

Comment:
I dont think we need a return value here. Its changed in place, i.e., inside the class object anyhow

Reply:
Mh, we're also instancing the class-object in `from_frequency_range()`, without the return we don't get any object to work with.

Reply:
ah, sorry - make sense!